### PR TITLE
Android vibration fixes.

### PR DESCRIPTION
**Pull request by MAJigsaw77**
* Fixed some deprecation warnings.
* Cancel the vibration when the app is being paused or destroyed.

### CODE REVIEW

**Comment by MAJigsaw77** [comment]
@player-03 Can i get a review on this pr?

File: templates/android/template/app/src/main/java/org/haxe/lime/GameActivity.java [hunk 1]
```diff
@@ -137,6 +139,14 @@ protected void onCreate (Bundle state) {
...
 	@Override protected void onDestroy () {
 
+		if (vibrator != null) {
+			
+			Log.d ("GameActivity", "Cancelling vibration");
+
+			vibrator.cancel ();
+
+		}
+
```
Comment (player-03) [inline templates/android/template/app/src/main/java/org/haxe/lime/GameActivity.java]: Is this necessary? [The documentation](https://developer.android.com/reference/android/os/Vibrator) claims ongoing vibrations will be stopped automatically.

Comment (MAJigsaw77) [reply templates/android/template/app/src/main/java/org/haxe/lime/GameActivity.java]: > Is this necessary? [The documentation](https://developer.android.com/reference/android/os/Vibrator) claims ongoing vibrations will be stopped automatically.

I'll remove it then

**Comment by player-03** [comment]
Interesting. According to the documentation, [the first entry in the array is the delay before vibration starts](https://developer.android.com/reference/android/os/VibrationEffect#createWaveform\(long[],%20int\)), so they recommend leaving that 0.

The question is, is this delay a bug or a feature? If it's a bug, we could fix it now. If it's a feature, we probably want to add an argument to toggle it on or off.

**Comment by MAJigsaw77** [comment]
> Interesting. According to the documentation, [the first entry in the array is the delay before vibration starts](https://developer.android.com/reference/android/os/VibrationEffect#createWaveform(long%5B%5D,%20int)), so they recommend leaving that 0.
> 
> The question is, is this delay a bug or a feature? If it's a bug, we could fix it now. If it's a feature, we probably want to add an argument to toggle it on or off.

I don't really know what to say tbh

**Pull request merged**